mypy/nodes.py

```diff
@@ -616,11 +617,13 @@ class FuncItem(FuncBase):
     def __init__(self,
                  arguments: List[Argument],
                  body: 'Block',
-                 typ: 'Optional[mypy.types.FunctionLike]' = None) -> None:
+                 typ: 'Optional[mypy.types.FunctionLike]' = None,
+                 num_pos_only: int = 0) -> None:
         super().__init__()
         self.arguments = arguments
         self.arg_names = [arg.variable.name for arg in self.arguments]
         self.arg_kinds: List[ArgKind] = [arg.kind for arg in self.arguments]
+        self.num_pos_only = num_pos_only
```

Comment:
This looks quite similar to `max_pos`. Would it be better named `min_pos_only`, for consistency (also is this the minimum number if some positional-only arguments have default values)? 

What about using `arg_kinds` + `arg_names` to calculate this? An argument with `ARG_POS` kind and `None` name is positional-only, I think. As long as the arguments have correct kinds, I think that this would do the right thing, and we wouldn't need the extra argument.  If this wouldn't work and we can't infer the value from `arguments`, please add a comment that explains it.